base/server/python/pki/server/instance.py

```diff
@@ -511,11 +511,27 @@ def get_sslserver_cert_nickname(self):
         # If not available, load SSL server cert nickname from serverCertNick.conf
         # TODO: Remove serverCertNick.conf
 
-        server_cert_nick_conf = os.path.join(self.conf_dir, 'serverCertNick.conf')
+        logger.info('Getting serverCertNickFile from server.xml')
+
+        document = etree.parse(self.server_xml, parser)
+        server = document.getroot()
+        server_cert_nick_conf = None
+
+        connectors = server.findall('Service/Connector')
+        for connector in connectors:
 
-        if not os.path.exists(server_cert_nick_conf):
+            if connector.get('secure') != 'true':
+                # not a secure connector -> skip
+                continue
+
+            server_cert_nick_conf = connector.get('serverCertNickFile')
+
+        if not server_cert_nick_conf:
+            # no secure connector -> no nickname
             return None
```

Comment:
I think the check should be inside the for loop and skip other connectors if you get the value.